dtfs-central-api/src/v1/controllers/utilisation-report-service/get-fee-record-details-by-id.controller/helpers.test.ts

```diff
@@ -0,0 +1,64 @@
+import { FeeRecordEntityMockBuilder, ReportPeriod, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { mapToFeeRecordDetails } from './helpers';
+import { getBankNameById } from '../../../../repositories/banks-repo';
+
+jest.mock('../../../../repositories/banks-repo');
+
+describe('get-fee-record-details-by-id.controller helpers', () => {
+  describe('mapToFeeRecordDetails', () => {
+    const bankId = '123';
+
+    const utilisationReport = new UtilisationReportEntityMockBuilder().withBankId(bankId).build();
+
+    it('returns an object containing the bank', async () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).build();
+
+      const bankName = 'Test bank';
+      jest.mocked(getBankNameById).mockResolvedValue(bankName);
+
+      // Act
+      const feeRecordDetails = await mapToFeeRecordDetails(feeRecord);
+
+      // Assert
+      expect(feeRecordDetails.bank).toEqual({ id: bankId, name: bankName });
+    });
+
+    it('returns an object containing the report period', async () => {
+      // Arrange
+      const reportPeriod: ReportPeriod = {
+        start: { month: 1, year: 2024 },
+        end: { month: 1, year: 2024 },
+      };
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).build();
+      feeRecord.report.reportPeriod = reportPeriod;
+
+      // Act
+      const feeRecordDetails = await mapToFeeRecordDetails(feeRecord);
+
+      // Assert
+      expect(feeRecordDetails.reportPeriod).toEqual(reportPeriod);
+    });
+
+    it('returns an object containing the mapped fee record details', async () => {
```

Comment:
This might need renaming or splitting as the mapping function is meant to return `fee record details` and this case is testing that all props apart from the bank and report period are present